### PR TITLE
Rank patterns when doing components decomposition

### LINKED ISSUE #613: Inference on decomposed class
Currently dataset is represented with class components instead of a single file. We have to do the inference the same way. Gathered ranks must be merged into a single list in the following way:  first goes all first elements, then all seconds and so on. Order on components is arbitrary. 

It might be also added to stat command.

### PR DESCRIPTION
**Pull request by lyriccoder**
1. fix #613 
Now we have components and we rank patterns in components. Interface has been changed, we need to changed `rank()` function also.

### CODE REVIEW

**Comment by lyriccoder** [comment]
1) What I see is that current LCOM4 is broken. Untill #615 is not fixed, this PR won't be merged

2) We need to discuss some certain options (Ignore setters, getters, constructors) for LCOM4

3) What should we do with Inner classes?

4) @yegor256 asked to count all unique patterns. We showed number of patterns with total number of patterns:

```
D:\target\0001\fast\Configuration.java  [225]: Non final class (P24: 9.40 1/3)
D:\target\0001\fast\Configuration.java  [225]: Implements Multi (P7: 2.70 2/3)
D:\target\0001\fast\Configuration.java  [3019]: Instance of (P8: 10.80 3/3)
```
Should we now group it by LCOM value, not by a file:
```
D:\target\0001\fast\Configuration.java 1 component  [225]: Non final class (P24: 9.40 1/2)
D:\target\0001\fast\Configuration.java 1 component   [225]: Implements Multi (P7: 2.70 2/2)
D:\target\0001\fast\Configuration.java 2nd component   [3019]: Instance of (P8: 10.80 1/1)
```

If not, in this case it is not useless to extract LCOM component, because ncss value is low for LCOM component and the patterns importance will be increased

**Review by aravij — CHANGES_REQUESTED** [review]

File: requirements.txt [hunk 1]
```diff
@@ -18,5 +18,4 @@ cached-property==1.2.0
 deprecated==1.2.10
 typing-extensions; python_version<'3.8'
 tqdm == 4.32.1
-bs4==0.0.1
-pebble==4.5.3
```
[the remark this answers is not in the document]
Comment (lyriccoder) [reply requirements.txt]: fixed, there was a bad merge, i think

**Review by aravij — APPROVED** [review]

**Comment by acheshkov** [comment]
@rultor merge

**Comment by acheshkov** [comment]
@rultor merge

**Comment by lyriccoder** [comment]
tenor
@rultor?

**Pull request merged**